src/scheduler/resource_resv.c

```diff
@@ -1004,25 +1009,35 @@ find_resource_req(resource_req *reqlist, resdef *def)
  * @param[out]	req	-	the resource_req to set
  * @param[in]	val -	the string value
  *
- * @return	nothing
- *
+ * @return	int
+ * @retval	1 for Success
+ * @retval	0 for Error
  */
-void
+int
 set_resource_req(resource_req *req, char *val)
 {
 	resdef *rdef;
+
 	/* if val is a string, req -> amount will be set to SCHD_INFINITY */
 	req->amount = res_to_num(val, &(req->type));
 	req->res_str = string_dup(val);
 
 	if (req->def != NULL)
 		rdef = req->def;
 	else {
-	rdef = find_resdef(allres, req->name);
+		rdef = find_resdef(allres, req->name);
 		req->def = rdef;
 	}
 	if (rdef != NULL)
 		req->type = rdef->type;
+
+	if (req->amount == SCHD_INFINITY) {
+		/* Verify that this is actually a non-numeric resource */
+		if (!(req->def->type.is_string))
```

Comment:
You don't necessarily need the inner set of ().  The -> and . operators have higher precedence than !